CQ-4292744 Integrate CQAssetsClient into aem-testing-clients

markjhood opened the pull request:
This PR integrates a new Direct Binary Access client (written by @bobvanmanen) into aem-testing-clients.  It provides a method  _uploadAsset_ to  create assets on Skyline environments, with a fallback to the Create Asset servlet on non-Skyline environments.  Methods are also provided to wait for asset processing and to check processing status.

Commit:
CQ-4292744 Integrate CQAssetsClient into aem-testing-clients

Commit:
Remove Metadata.java

markjhood commented:
I think it would be convenient to have a method _waitAssetProcessed(String assetPath, long timeout, long delay)_ in addition to the existing  _waitAssetProcessed(String assetPath)_ which applies a timeout of 30000ms and a delay of 500ms.  The current values are reasonable for most applications I think, and it's true that you could code a custom version of _waitAssetProcessed(String assetPath)_ from the existing methods, but the method would provide more flexibility with a minimum of effort.  What do you think?

volteanu commented on src/main/java/com/adobe/cq/testing/client/assets/Util.java:
```diff
@@ -0,0 +1,127 @@
+     * @return Text value in field or null if the field doesn't exist
+     */
+    // TODO: Contribute to org.apache.sling.testing.clients.util.JsonUtils
+    public static String getJsonTextValue(JsonNode parent, String fieldName) {
```
wouldn't `JsonNode#path` help here?

markjhood replied:
Yes, no need for getJsonTextValue().  Done.

volteanu reviewed (approved):
Thanks @markjhood , the code looks good to me and inline with the framework! I just noticed that some headers need to be updated or added. Would you take care or should I?

markjhood commented:
Talked to Bob van Manen, I am going to add waitAssetProcessed(String assetPath, long timeout, long delay) before merging and also take a look at your suggestion for improvements.

Commit:
CQ-4292744 Add waitAssetProcessed(String assetPath, long timeout,
long delay).  Use JsonNode#path() instead of Util.getJsonTextValue().

Commit:
CQ-4292744 Add/update headers for Apache License

markjhood commented:
Added waitAssetProcessed(String assetPath, long timeout, long delay).  
Used JsonNode#path() instead of Util.getJsonTextValue(). 
Added/updated code headers for Apache License 2.0.

Not sure why Codecov popped in there.  Let me know if there is anything I need to do.  Otherwise, the code is ready for merging.

volteanu commented:
Thanks again @markjhood ! codecov is checking coverage, which is slightly decreasing. It's usually good to have unittests for the new code, but we're currently very low with the coverage so we don't make this blocking.
